### DISCUSSION TITLE
Repeater does not get updated when Data Provider is refreshed

### DISCUSSION THREAD

**Discussion by JohnM053** [Budibase/budibase#13471] [Help] [answered]
Hi folks, 
for a few days now, I've been trying to solve the following issue. I'm not sure if I'm doing something wrong or if it's a bug. In my application, a Google Sheet is connected with Budibase. Users can make changes to an entry in a side panel. However, before the changes are saved, I want to check if the entry has not been blocked in the meantime (meaning the time between calling up the entry and opening the side panel and pressing the Save button). This is indicated by the word "BLOCKED" in a specific column. If it has been blocked, the changes should not be saved.

Here's what I've done so far: As an on-click action of the Save button, the Data Provider should first be refreshed, and then using Continue-if/Stop-if function calling the Repeater, it should be checked whether the entry has been blocked in the meantime. The problem is that although the Data Provider is refreshed, the Repeater still returns the old value (i.e., before the Data Provider was refreshed). That means that even if the entry has been blocked in the meantime, the row gets saved.

Has anyone else encountered this issue? Or do you have a workaround for this?

![Unbenannt 1](https://github.com/Budibase/budibase/assets/166725786/57c102a9-ce0f-4a26-aee2-ae2ecea3db27)</div>

**Comment by aptkingston** [comment, marked as answer]
Hey @JohnM053.

I think the issue here is that the "refresh data provider" action is not synchronous, so your following actions will continue before the refresh is actually completed. We could potentially add a new option to that action, so that you can choose if you want to wait for the data to be refreshed or not. Feel free to open a feature request for this if that sounds like something you'd be interested in.

There are 2 other solutions that come to mind, but both are paid features:
- You could use the new "auto refresh" setting on your data provider to automatically refresh it every 5 seconds. That way you'll be able to show that it's blocked automatically, without needing to check here in your action. (There might also be a plugin that does this for free). There is still a very small window (5 seconds) where it could be blocked before pressing save, so maybe you'd prefer the next option.
- You could use a synchronous automation to query the item, then return whether or not it's blocked. This is all sync, so the actions will wait for the response and you'll be able to act on it appropriately.

**Reply by JohnM053** [reply]
Thank you very much, that helped a lot!

**Comment by aptkingston** [comment]
I've closed the issue and we can handle this in discussions here.
